php-checks/src/main/java/org/sonar/php/checks/CountInsteadOfEmptyCheck.java

```diff
@@ -91,6 +92,21 @@ private boolean isCountFunction(FunctionCallTree tree) {
     return FUNCTION_PREDICATE.test(TreeValues.of(tree, context().symbolTable()));
   }
 
+  private static final Tree.Kind[] USED_STATEMENTS = {
+    Tree.Kind.EXPRESSION_STATEMENT,
+    Tree.Kind.RETURN_STATEMENT,
+    Tree.Kind.IF_STATEMENT,
+    Tree.Kind.WHILE_STATEMENT,
+    Tree.Kind.DO_WHILE_STATEMENT,
+    Tree.Kind.FOR_STATEMENT,
+    Tree.Kind.FOREACH_STATEMENT,
+    Tree.Kind.SWITCH_STATEMENT,
+    Tree.Kind.ASSIGNMENT,
+    Tree.Kind.FUNCTION_DECLARATION,
+    Tree.Kind.METHOD_DECLARATION};
+
+  private static final Tree.Kind[] AND_OR_TREE = {Tree.Kind.CONDITIONAL_AND, Tree.Kind.CONDITIONAL_OR, Tree.Kind.ALTERNATIVE_CONDITIONAL_AND, Tree.Kind.ALTERNATIVE_CONDITIONAL_OR};
```

Comment:
I think this is not a tree, hence, we should not call it a tree 😉.
How about:
```suggestion
  private static final Tree.Kind[] AND_OR_KINDS = {Tree.Kind.CONDITIONAL_AND, Tree.Kind.CONDITIONAL_OR, Tree.Kind.ALTERNATIVE_CONDITIONAL_AND, Tree.Kind.ALTERNATIVE_CONDITIONAL_OR};
```